nav2_navfn_planner/src/main.cpp

```diff
@@ -15,14 +15,22 @@
 #include <memory>
 
 #include "nav2_navfn_planner/navfn_planner.hpp"
-#include "rclcpp/rclcpp.hpp"
+#include "nav2_util/executors.hpp"
 
 int main(int argc, char ** argv)
 {
   rclcpp::init(argc, argv);
-  auto node = std::make_shared<nav2_navfn_planner::NavfnPlanner>();
-  rclcpp::spin(node->get_node_base_interface());
-  rclcpp::shutdown();
 
+  auto planner = std::make_shared<nav2_navfn_planner::NavfnPlanner>();
+
+  // The planner needs a special executor
```

Comment:
Feedback: I'd just put in the multithreaded executor, making this wrapper makes it less clear to me than just having it all together. 

If there's issues with the planner as a service, maybe it should be an action, but in general I dont think a multi-threaded spinner is something to be avoided, I make heavy use of them in autonomy systems in ROS1 to allow for a bunch of services/callbacks to be triggered without waiting.